crates/bevy_render/macros/src/as_bind_group.rs

```diff
@@ -200,25 +200,40 @@ pub fn derive_as_bind_group(ast: syn::DeriveInput) -> Result<TokenStream> {
                     let StorageAttrs {
                         visibility,
                         read_only,
+                        buffer,
                     } = get_storage_binding_attr(nested_meta_items)?;
                     let visibility =
                         visibility.hygenic_quote(&quote! { #render_path::render_resource });
 
                     let field_name = field.ident.as_ref().unwrap();
                     let field_ty = &field.ty;
 
-                    binding_impls.push(quote! {{
-                        use #render_path::render_resource::AsBindGroupShaderType;
-                        let mut buffer = #render_path::render_resource::encase::StorageBuffer::new(Vec::new());
-                        buffer.write(&self.#field_name).unwrap();
-                        #render_path::render_resource::OwnedBindingResource::Buffer(render_device.create_buffer_with_data(
-                            &#render_path::render_resource::BufferInitDescriptor {
-                                label: None,
-                                usage: #render_path::render_resource::BufferUsages::COPY_DST | #render_path::render_resource::BufferUsages::STORAGE,
-                                contents: buffer.as_ref(),
-                            },
-                        ))
-                    }});
+                    let min_binding_size = if buffer {
+                        quote! {None}
```

Comment:
Does `Buffer` not implement `ShaderType`?

Reply:
No, it doesn't implement it